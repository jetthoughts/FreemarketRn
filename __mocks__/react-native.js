const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
