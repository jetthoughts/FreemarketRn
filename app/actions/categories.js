export const loadCategories = () => ({ type: 'LOAD_CATEGORIES' });
export const selectCategory = (id) => ({ type: 'SELECT_CATEGORY', id });
