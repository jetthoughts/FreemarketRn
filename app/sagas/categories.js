import { put, call, takeLatest } from 'redux-saga/effects';

import {
  categoriesSuccess,
  categoriesError,
  requestCategories,
} from '../reducers/categories';
import { fetchAllCategories } from '../Api';

export function* fetchCategories() {
  try {
    const response = yield call(fetchAllCategories);
    const [, ...categories] = response.data;
    yield put(categoriesSuccess(categories));
  } catch (e) {
    yield put(categoriesError(e));
  }
}

export function* watchCategoriesRequest() {
  yield takeLatest(requestCategories.getType(), fetchCategories);
}
