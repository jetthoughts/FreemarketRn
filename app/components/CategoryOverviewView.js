import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import CategoryHeaderAndroid from './CategoryHeader.android';

const CategoryOverviewView = ({ category, platform, openDrawer }) => (
  <View>
    {platform === 'android' ? <CategoryHeaderAndroid category={category} openDrawer={openDrawer} /> : null}
  </View>
);


CategoryOverviewView.propTypes = {
  category: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
  }).isRequired,
  platform: PropTypes.oneOf(['android', 'ios']).isRequired,
};

export default CategoryOverviewView;
