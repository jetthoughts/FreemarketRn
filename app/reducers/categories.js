import { combineReducers } from 'redux';
import { createAction, createReducer } from 'redux-act';
import categories from '../constants/categories';

export const loadCategories = createAction('load categories list');
export const selectCategory = createAction('change current category');

const listReducer = createReducer({
  [loadCategories]: (state) => categories,
}, []);

const currentReducer = createReducer({
  [selectCategory]: (state, payload) => payload,
}, 0)

export default combineReducers({
  list: listReducer,
  current: currentReducer,
});
