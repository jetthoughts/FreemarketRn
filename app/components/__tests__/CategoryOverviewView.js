import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CategoryOverviewView from '../CategoryOverviewView';
import categories from '../../constants/categories';

it('renders correctly', () => {
  const tree = renderer.create(
    <CategoryOverviewView category={categories[0]} platform="android" />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
