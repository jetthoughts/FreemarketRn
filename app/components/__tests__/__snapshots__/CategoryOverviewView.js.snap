// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Icon.Button
    navIconName="bars"
    onIconClicked={[Function]}
    style={
      Object {
        "backgroundColor": "dodgerblue",
        "height": 56,
      }
    }
    title="Antiques"
    titleColor="white"
  />
</View>
`;
